### PR TITLE
Document indexation extensibility

### PR DESCRIPTION
**Pull request by mmetayer**
**Description (for Contributor and Core Developer)**

Adds documentation about extensibility of indexation for custom attribute types

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: internal_doc/enrichment/add_new_attribute_type.md [hunk 1]
```diff
@@ -153,15 +140,6 @@ class RangeValueFactory implements ValueFactory
...
-    acme.range.factory.write.value.range:
-        class: Acme\RangeBundle\Product\Factory\Write\Value\RangeValueFactory
-        public: false
-        arguments:
-            - 'Akeneo\Pim\Enrichment\Component\Product\Value\ScalarValue'
-            - 'range'
-        tags:
-            - { name: pim_catalog.factory.value }
-
     acme.range.factory.value.range:
```
Comment (pierallard) [inline internal_doc/enrichment/add_new_attribute_type.md]: The new way is not to put it in the Read namespace ? (not sure)

File: internal_doc/enrichment/add_new_attribute_type.md [hunk 2]
```diff
@@ -225,7 +203,106 @@ class RangeComparator implements ComparatorInterface
...
+          "min": 0,
+          "max": 50
+        }
+      }
+    }
+  } 
+}
+```
+
+For this you will just need to create the adequate normalizer:
```
Comment (pierallard) [inline internal_doc/enrichment/add_new_attribute_type.md]: remove "just"

File: internal_doc/enrichment/add_new_attribute_type.md [hunk 3]
```diff
@@ -225,7 +203,106 @@ class RangeComparator implements ComparatorInterface
...
+                mapping:
+                    type: 'double'
+        -
+            range_max:
+                path_match: 'values.*-range.*.max'
+                mapping:
+                    type: 'double'
+```
+
+All you have to do now is include this file in the elasticsearch configuration:
```
Comment (pierallard) [inline internal_doc/enrichment/add_new_attribute_type.md]: Warning, this documentation does not provides the UI code to add a filter to a datagrid.

File: internal_doc/enrichment/add_new_attribute_type.md (hunk 1 above)
Comment (ahocquard) [reply internal_doc/enrichment/add_new_attribute_type.md]: Read namespace should disappear, it's part of the work that left to do

**Pull request merged**